Automattic/wp-calypso
# Sites: Remove Jetpack minimum version functionality

Jetpack 3.3 is the minimum version that contained the "Manage" module, allowing Calypso to work with Jetpack sites. It was released **5 years ago**, and the current version is 8.0. Jetpack's policy is to support current version - 1, so it's about time we remove the obsolete checks that assure the site has Jetpack 3.3 or newer. 

This PR removes all of that functionality:

* `NonSupportedJetpackVersionNotice` component that was used in plugins management.
* `site.hasMinimumJetpackVersion` and its usages.
* `siteHasMinimumJetpackVersion` selector and its usages.
* The `jetpack_min_version` config var.

#### Changes proposed in this Pull Request

* Plugins: Remove usages of `NonSupportedJetpackVersionNotice`
* Plugins: Remove `NonSupportedJetpackVersionNotice`
* Framework: Remove any occurrence of `site.hasMinimumJetpackVersion`
* State: Remove usage of `siteHasMinimumJetpackVersion`
* State: Delete `siteHasMinimumJetpackVersion` selector
* State: Remove usages of `jetpack_min_version` config var
* State: Remove an obsolete test
* Config: Remove `jetpack_min_version` complete

#### Testing instructions

* Checkout this branch.
* Smoke test everything with Jetpack sites in Calypso.
* There should be no changes, everything should work well.
* Verify tests still pass.
